framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/AClassWrapper.java

```diff
@@ -0,0 +1,191 @@
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.javacutil.BugInCF;
+import scenelib.annotations.Annotation;
+import scenelib.annotations.el.AClass;
+import scenelib.annotations.util.JVMNames;
+
+/**
+ * A wrapper for the AClass class from scene-lib that carries additional information that is useful
+ * during WPI.
+ *
+ * <p>This might be better as a subclass of AClass, but AClass is final.
+ */
+public class AClassWrapper {
+
+    /** The wrapped AClass object. */
+    private AClass theClass;
+
+    /**
+     * The methods of the class. Keys are the names of methods, entries are AMethodWrapper objects.
+     * Mirrors the "methods" field of AClass.
+     */
+    private final Map<String, AMethodWrapper> methods = new HashMap<>();
+
+    /**
+     * The fields of the class. Keys are the names of the fields, entries are AFieldWrapper objects.
+     * Wraps the "fields" field of AClass.
+     */
+    private final Map<String, AFieldWrapper> fields = new HashMap<>();
+
+    /** The enum constants of the class, or null if this class is not an enum. */
+    private @MonotonicNonNull List<VariableElement> enumConstants = null;
+
+    /**
+     * The type element representing the class. Clients must call {@link
+     * #setTypeElement(TypeElement)} before interacting with this field.
+     */
+    private @MonotonicNonNull TypeElement typeElement = null;
+
+    /**
+     * Wrap an AClass.
+     *
+     * @param theClass the wrapped object
+     */
+    public AClassWrapper(AClass theClass) {
+        this.theClass = theClass;
+    }
+
+    /**
+     * Call before doing anything with a method. Interacts with scenelib and fetches or creates an
+     * appropriate AMethodWrapper object.
+     *
+     * <p>Results are interned.
+     *
+     * @param methodElt the executable element representing the method
+     * @return an AMethodWrapper representing the method
+     */
+    public AMethodWrapper vivifyMethod(ExecutableElement methodElt) {
+        String methodName = JVMNames.getJVMMethodName(methodElt);
+        if (methods.containsKey(methodName)) {
+            return methods.get(methodName);
+        } else {
+            AMethodWrapper wrapper =
+                    new AMethodWrapper(
+                            theClass.methods.getVivify(methodName), methodElt.getReturnType());
+            methods.put(methodName, wrapper);
+            return wrapper;
+        }
+    }
+
+    /**
+     * Get all the methods that have been vivified (that is, interacted with) on a class.
```

Comment:
Wording "interact with" again.  (I'm giving each comment once in general, but they may apply to multiple places in the code.